### PR TITLE
austin.intermediate.summer2.sears.charlie

### PR DESCRIPTION
**Pull request by inlandrecords**
I changed the false to true for lesson 0 test

### CODE REVIEW

File: homework/01LessonOne.js [hunk 1]
```diff
@@ -17,27 +17,27 @@ var myUndefined;
 //
 // What is the difference between null and undefined?
```
Comment (tjefferson08) [inline homework/01LessonOne.js:18]: Snuck a question in here! Could you take a shot at an answer in a comment underneath?

File: homework/01LessonOne.js [hunk 2]
```diff
@@ -105,8 +106,9 @@ var notNotTrue;
 //
 // What is the difference between the "and" and the "or" operators? Explain why you would use each of them.
 //
-// Your Answer Goes Here:
-//
+// Your Answer Goes Here: 'And' requires both parts of a problem to be true to
```
Comment (tjefferson08) [inline homework/01LessonOne.js:111]: Awesome! Vocabulary lesson: replace `problem` with `expression` to sound extra fancy.

File: apps/01PigLatin.js [hunk 3]
```diff
@@ -7,8 +7,37 @@ prompt.start();
...
+var firstPart = word.slice(0, vowelIndex);
+var restWord = word.slice(vowelIndex, word.length);
+
+return restWord + firstPart + 'ay';
+
+console.log( pigLatin('egg') );
+if vowelIndex(0)
+  return word + 'yay'
+
+console.log( 'HeLlO'.toLowerCase() );
```
Comment (paulrenenichols) [inline apps/01PigLatin.js]: This looks incomplete. Did you try implementing this while watching the [lecture video](https://www.youtube.com/watch?v=d8hF0AMiDL4)?

The assignment is to replace your apps/01PigLatin.js file with this code https://gist.github.com/paulrenenichols/099c38009a569d603a098060f6cc3018, then fill out the concept checks.

Comment (inlandrecords) [inline apps/01PigLatin.js]: I'm currently trying to push my changes to git. My machine says its working but I don't see the changes here.

File: apps/01PigLatin.js [hunk 4]
```diff
@@ -7,11 +7,82 @@ prompt.start();
...
+  word  = word.toLowerCase();
 
+  // Concept Check
+  //
+  // What is the significance of setting vowelIndex to -1?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+  //
+  // Your Answer:
+  //We are setting our code to assume there are no vowels.
```
Comment (tjefferson08) [inline apps/01PigLatin.js:29]: Yup

File: apps/01PigLatin.js [hunk 5]
```diff
@@ -7,11 +7,82 @@ prompt.start();
...
+      vowelIndex = word.indexOf('y');
+  }
+
+  // Concept Check
+  //
+  // Why are we using String.prototype.slice() here? What role does vowelIndex play?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice
+  //
+  // Your Answer: We are using the slice to slice the word. We need the vowelIndex
```
Comment (tjefferson08) [inline apps/01PigLatin.js:62]: What the heck does slice mean? Does it break the original variable or is it OK?

File: apps/02RockPaperScissors.js [hunk 6]
```diff
@@ -8,9 +8,20 @@ prompt.start();
 function rockPaperScissors(hand1, hand2) {
 
     // Write code here
-
+if (hand1 === hand2)
+  {return "It's a tie!"}
```
Comment (tjefferson08) [inline apps/02RockPaperScissors.js:12]: MOAR WHITESPACE YO

File: homework/02LessonTwo.js [hunk 7]
```diff
@@ -34,17 +44,34 @@ function equals(argument1, argument2) {
...
 // Problem 4:
 // write a function named addThree(number1, number2, number3)
 // this function will add three numbers
 // you must call your function add() in addThree()
 
+function addThree(number1, number2, number3)
+  {
+    var firstAdd = add(number1, number2);
+
+    return firstAdd + number3;
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:72]: can you use add again instead of `+` ?

File: homework/02LessonTwo.js [hunk 8]
```diff
@@ -54,22 +81,29 @@ function lessThanOrEqualTo(parameter1, parameter2) {
 // What is a function? How do you define a function in Javascript?
 //
 // Your Answer Goes Here:
-//
+//A function is code that takes one or more arguments and returns a new value
+//based on what the function is set up to do. The way you define a function is
+//by stating 'function_space_nameoffunction(argument1, argument2)'
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:86]: Nice

File: homework/02LessonTwo.js [hunk 9]
```diff
@@ -54,22 +81,29 @@ function lessThanOrEqualTo(parameter1, parameter2) {
...
+//Following this code you can add if statements and conditionals.
 //
 // What is a return value?
 //
 // Your Answer Goes Here:
-//
+//A return value is what the function outputs at the end of a function.
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:92]: Does it _have_ to be at the end?

File: homework/02LessonTwo.js [hunk 10]
```diff
@@ -54,22 +81,29 @@ function lessThanOrEqualTo(parameter1, parameter2) {
...
 //
 // How do you define a named function?
 //
-// Your Answer Goes Here:
+// Your Answer Goes Here:  A named function is one action you want your code to
+// do. You name the function whatever you like and then you name two parameters and
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:97]: Doesn't HAVE to be two params. Can be zero, one, two, or a BILLION

File: homework/02LessonTwo.js [hunk 11]
```diff
@@ -54,22 +81,29 @@ function lessThanOrEqualTo(parameter1, parameter2) {
...
+// tell the fuction what to do with those parameters.
 //
 //
 // What is a parameter? What is an argument? Is there a difference between the two?
 //
 // Your Answer Goes Here:
-//
+//A parameter is a data point or possibly user input that your function will
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:104]: Good good

File: homework/02LessonTwo.js [hunk 12]
```diff
@@ -126,9 +220,9 @@ function whichSpecies(character) {
 //
 // In your own words, explain what conditionals do.
 //
-// Your Answer Goes Here:
-//
-//
+// Your Answer Goes Here: They turn raw user data into what the programmer
```
Comment (tjefferson08) [inline homework/02LessonTwo.js:223]: Not bad!

File: apps/01PigLatin.js (hunk 5 above)
Comment (inlandrecords) [inline apps/01PigLatin.js:62]: it cuts it into two pieces like a katana or machete.

File: apps/02RockPaperScissors.js (hunk 6 above)
Comment (inlandrecords) [inline apps/02RockPaperScissors.js:12]: I'm working on this. I did better with homework 2.

File: homework/02LessonTwo.js (hunk 7 above)
Comment (inlandrecords) [inline homework/02LessonTwo.js:72]: I wasn't sure about this one.

File: apps/02RockPaperScissors.js [hunk 13]
```diff
@@ -8,9 +8,20 @@ prompt.start();
...
+if (hand1 === 'scissors' && hand2 === 'paper')
+  {return "Hand one wins!";}
+if (hand1 === 'rock' && hand2 === 'scissors')
+    {return "Hand one wins!";}
+if (hand1 === 'paper' && hand2 === 'rock')
+    {return "Hand one wins!";}
+  else
+    {return "Hand two wins!";}
```
Comment (tjefferson08) [inline apps/02RockPaperScissors.js:20]: Formatting is a little wonky, but nice, simple if/else blocks. Good work

File: homework/03LessonThree.js [hunk 14]
```diff
@@ -20,16 +20,17 @@ var firstName = 'susan';
...
 // Problem 2:
 // What is the value of lastName at this point in the file?
-var valueOfLastName;
+var valueOfLastName = 'plaid';
 
 // Problem 3:
 // Fix the following function so that it returns the first argument
+var firstArgument = 'bob';
 function returnTheFirstArgument(firstArgument) {
-  var firstArgument = 'bob';
```
Comment (tjefferson08) [inline homework/03LessonThree.js:32]: noice!

File: homework/03LessonThree.js [hunk 15]
```diff
@@ -41,7 +42,9 @@ function returnTheFirstArgument(firstArgument) {
 // What is scope?
 //
 // Your Answer Goes Here:
-//
+//It is the area of the code that contains the arguments and variables a person
```
Comment (tjefferson08) [inline homework/03LessonThree.js:45]: oooo a bucket! nice.

File: homework/03LessonThree.js [hunk 16]
```diff
@@ -57,19 +60,19 @@ var ternaryResult4 = ""     ? 'first' : 'second';
 
 // Problem 4:
 // What is the value of ternaryResult1
-var problemFourAnswer;
+var problemFourAnswer = true ? 'first' : 'second';
```
Comment (tjefferson08) [inline homework/03LessonThree.js]: Let's talk about this ^ it passes the test, but it's not how we wanted you to answer. We wanted you to 'choose' an answer like this

``` javascript
var problemFourAnswer = 'some answer';
```

File: homework/03LessonThree.js [hunk 17]
```diff
@@ -57,19 +60,19 @@ var ternaryResult4 = ""     ? 'first' : 'second';
...
 
 // Problem 5:
 // What is the value of ternaryResult2
-var problemFiveAnswer;
+var problemFiveAnswer = false ? 'first' : 'second';
```
Comment (tjefferson08) [inline homework/03LessonThree.js]: Ditto here

File: homework/03LessonThree.js [hunk 18]
```diff
@@ -57,19 +60,19 @@ var ternaryResult4 = ""     ? 'first' : 'second';
...
 
 // Problem 6:
 // What is the value of ternaryResult3
-var problemSixAnswer;
+var problemSixAnswer = 4      ? 'first' : 'second';
```
Comment (tjefferson08) [inline homework/03LessonThree.js]: Ditto here

File: homework/03LessonThree.js [hunk 19]
```diff
@@ -57,19 +60,19 @@ var ternaryResult4 = ""     ? 'first' : 'second';
...
 
 // Problem 7:
 // What is the value of ternaryResult4
-var problemSevenAnswer;
+var problemSevenAnswer = ""     ? 'first' : 'second';
```
Comment (tjefferson08) [inline homework/03LessonThree.js]: annnnnnd here

File: homework/03LessonThree.js [hunk 20]
```diff
@@ -80,7 +83,10 @@ var problemSevenAnswer;
 //
 // Your Answer Goes Here:
 //
-//
+//The ternary operator (?) is like a toggle. It outputs one result and then
```
Comment (tjefferson08) [inline homework/03LessonThree.js]: Ternaries are for more than toggling! Let's update this answer with a little more generic description

Read this! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

File: homework/03LessonThree.js [hunk 21]
```diff
@@ -89,17 +95,19 @@ var problemSevenAnswer;
 
 // Problem 8:
 // Create an empty array
-var emptyArray;
+var emptyArray = [ ];
 
 // Problem 9:
 // Create an array with 5 elements in it
-var lengthFiveArray;
+var lengthFiveArray = [ "Yoda", "Luke", "Leia", "Darth Vader", "Obi-Wan"];
```
Comment (tjefferson08) [inline homework/03LessonThree.js:104]: Sweet

File: homework/03LessonThree.js [hunk 22]
```diff
@@ -89,17 +95,19 @@ var problemSevenAnswer;
...
 
 // DO NOT MODIFY
 var nameArray = ['bob', 'fred', 'susan'];
 
 // Problem 10:
 // Replace the value 'fred' in nameArray with 'george'
+nameArray[2] = george;
```
Comment (tjefferson08) [inline homework/03LessonThree.js]: ahhh! needs quotes!

Comment (inlandrecords) [inline homework/03LessonThree.js]: fixed this error in latest commit <---- that sounds prettttay programmer-y. So proud of my newfound jargon skillz.

File: homework/03LessonThree.js (hunk 16 above)
Comment (inlandrecords) [inline homework/03LessonThree.js]: fixed all these in latest commit.

File: apps/03TicTacToe.js [hunk 23]
```diff
@@ -23,24 +23,84 @@ function printBoard() {
 
 function horizontalWin() {
     // Your code here
+    if (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn)
```
Comment (tjefferson08) [inline apps/03TicTacToe.js]: looks nice and clean, but we DEFINITELY want to wrap the bodies of our if/else statements in curly braces `{}`

File: apps/03TicTacToe.js [hunk 24]
```diff
@@ -23,24 +23,84 @@ function printBoard() {
...
+
+    }
+    else if (diagonalWin(true)) {
+
+      console.log('Player ' + playerTurn + 'Won!')
+
+    }
+  
+    else {
+      return false
```
Comment (tjefferson08) [inline apps/03TicTacToe.js]: Interesting! Do we ever want to return true?

File: apps/03TicTacToe.js [hunk 25]
```diff
@@ -23,24 +23,84 @@ function printBoard() {
...
 function ticTacToe(row, column) {
     // Your code here
+
+    board[row][column] = playerTurn;
+
+    playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+
 }
 
+checkForWin()
```
Comment (tjefferson08) [inline apps/03TicTacToe.js]: Why does this guy live all alone out here? _When_ do we ask for win?

File: homework/04LessonFour.js [hunk 26]
```diff
@@ -14,45 +14,75 @@ var numberArray = [1, 2, 3, 4, 5];
...
-var numberArrayAfterUnshift = [];
+var numberArrayAfterUnshift = [0, 1, 2, 3, 4, 5];
 
 // Problem 5:
 // Use pop() and shift() to make the array bravestWarriors look like this:
 // ['Danny', 'Chris', 'Beth', 'Wallow']
+
 var bravestWarriors = ['Catbug', 'Danny', 'Chris', 'Beth', 'Wallow', 'Impossibear'];
 
+var Impossibear = bravestWarriors.pop()
```
Comment (tjefferson08) [inline homework/04LessonFour.js:37]: cool. we don't use these variables, but I like it

File: homework/04LessonFour.js [hunk 27]
```diff
@@ -78,6 +112,8 @@ var nestedObjects = {
 };
 // update the starring property to have the value 'Tim Robbins'
 
+nestedObjects.users['bob roberts'].starring= 'Tim Robbins';
```
Comment (tjefferson08) [inline homework/04LessonFour.js:115]: Nice, I like that you favored dot notation. Easier to read

File: homework/04LessonFour.js [hunk 28]
```diff
@@ -78,6 +112,8 @@ var nestedObjects = {
...
+
 
 // ****
 // Concept Checkpoint
```
Comment (tjefferson08) [inline homework/04LessonFour.js:119]: Whoops! Fill this answer out for full points brotha

File: apps/04TowersOfHanoi.js [hunk 29]
```diff
@@ -16,25 +16,75 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-    // Your code here
+function movePiece(startStack, endStack) {
+
+  var piece = stacks[startStack].pop()
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:21]: Smooth

File: apps/04TowersOfHanoi.js [hunk 30]
```diff
@@ -16,25 +16,75 @@ function printStacks() {
...
+  stacks[endStack].push(piece);
 
 }
 
-function isLegal() {
-    // Your code here
+function isLegal(startStack, endStack) {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:26]: This function is beautiful

File: apps/04TowersOfHanoi.js [hunk 31]
```diff
@@ -16,25 +16,75 @@ function printStacks() {
...
+
+      return startStackArray[startStackArray.length - 1] < endStackArray[endStackArray.length - 1];
+
+  }
+
+  }
 
 function checkForWin() {
     // Your code here
+    if (stacks.c.length === 4)   {
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:55]: this looks a lot like the next `if` block, could we combine them somehow? Maybe with an OR ?

File: apps/04TowersOfHanoi.js [hunk 32]
```diff
@@ -16,25 +16,75 @@ function printStacks() {
...
 
 function towersOfHanoi(startStack, endStack) {
-    // Your code here
 
-}
+
+
+// Your code here
+
+  movePiece(startStack, endStack);
```
Comment (tjefferson08) [inline apps/04TowersOfHanoi.js:85]: This looks a little off to me. Do we always move the piece? Why'd we write that isLegal function?

File: homework/05LessonFive.js [hunk 33]
```diff
@@ -63,15 +63,16 @@ for (forLoopCounter = 0; forLoopCounter > -10; forLoopCounter--) {
...
 
 // ****
 // Concept Checkpoint
 //
 // Write your answer in comments
 //
 // What are loops? Why do we use them?
-//
+// Loops are a way to repeat an action (or function) several times without
+// having to rewrite code over and over again.
```
Comment (tjefferson08) [inline homework/05LessonFive.js:75]: Noice

File: apps/05Mastermind.js [hunk 34]
```diff
@@ -20,20 +22,60 @@ function generateSolution() {
...
+        correctLetterLocations++;
+        solutionArray[i] = null;
+}
 }
+  var correctLetters = 0;
 
-function mastermind(guess) {
-    // your code here
+  for (i = 0; i < solutionArray.length; i++) {
+    var targetindex = (guessArray[i] === solutionArray.indexOf())
```
Comment (tjefferson08) [inline apps/05Mastermind.js:48]: Hmm this indexOf() call doesn't look complete to me. Remember when we call `arr.indexOf(el)` we're looking for `el` somewhere inside `arr`

File: homework/06LessonSix.js [hunk 35]
```diff
@@ -89,15 +101,16 @@ var conferenceRoom = new ConferenceRoom();
...
 
 // ****
 // Concept Checkpoint
 //
 // Write your answer in comments
 //
 // What is a method?
-//
-//
+//A method is a reusable function that can be applied to Objects and object literals
```
Comment (tjefferson08) [inline homework/06LessonSix.js:112]: reuseable and guaranteed to be on every instance!

Comment (tjefferson08) [inline homework/06LessonSix.js:112]: reusable and guaranteed to be on every instance!

File: apps/06SpaceTravelToMars.js [hunk 36]
```diff
@@ -2,14 +2,60 @@
...
+
+function Ship(name, type, ability) {
+  this.name = name;
+  this.type = type;
+  this.ability = ability;
+  this.crew = [];
+
+  this.missionStatement = function() {
+  for  (var i = 0; i < this.crew.length; i++) {
+ var currentJob = this.crew[i].job;
```
[the remark this answers is not in the document]
Comment (tjefferson08) [reply apps/06SpaceTravelToMars.js:32]: Good variable use and good name choice! Bad indentation though :D

File: apps/06SpaceTravelToMars.js [hunk 37]
```diff
@@ -2,14 +2,60 @@
...
+//     else {
+//       return "Can't perform a mission yet.";
+// }
+}
+//This is the end of the for loop^^
+
+  return "Can't perform a mission yet.";
+
+}
+//This is the end of the missionStatement function^^
```
Comment (tjefferson08) [inline apps/06SpaceTravelToMars.js:56]: Nice comment! Proper indentation would be even better!

File: homework/07LessonSeven/script.js [hunk 38]
```diff
@@ -69,20 +71,30 @@ $(document).on('ready', function() {
...
   // Problem 12:
   // select the element with id="data-element"
   // and set its text (.text()) to the value of the data-content attribute
+var $dataElem = $('#data-element');
+var content = $dataElem.data('content');
+$dataElem.text(content);
+
+
 });
+//^^LEAVE THAT SHIZ ALON
```
[the remark this answers is not in the document]
Comment (tjefferson08) [reply homework/07LessonSeven/script.js:100]: Always a good idea

File: homework/07LessonSeven/script.js [hunk 39]
```diff
@@ -69,20 +71,30 @@ $(document).on('ready', function() {
...
   // you can do this by calling .data() on the selected element
-  window.dataIdValue;
+window.dataIdValue = $('#data-element').data('id');
+
 
   // Problem 12:
   // select the element with id="data-element"
   // and set its text (.text()) to the value of the data-content attribute
+var $dataElem = $('#data-element');
+var content = $dataElem.data('content');
```
Comment (tjefferson08) [inline homework/07LessonSeven/script.js:95]: Good intermediate variable here 👍

**Pull request closed without merging**